uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -451,28 +451,41 @@ public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDa
    *   And array of values from the Leave Request Status OptionGroup
    * @param bool $publicHolidaysOnly
    *   When true, will get the balance change only for the Public Holiday Leave Requests
+   * @param bool $expiredOnly
+   *   When true, will only count the balance change for expired days
    *
    * @return array
    */
   public static function getBalanceChangeByAbsenceType(
     $contactID,
     $periodID,
     $leaveRequestStatus = [],
-    $publicHolidaysOnly = false
+    $publicHolidaysOnly = false,
+    $expiredOnly = false
```

Comment:
Wondering if expired param should actually be on this method. After going through the method and the tests, its obvious that when the expiredOnly param is true, the method actually returns expired toil leave requests as well as expired brought forward entitlements which are not leave requests or related to one.
The method for returning expired balance changes might actually be well suited for the LeavePeriodEntitlement BAO. (just my thought) which is just the `getLeaveRequestBalance()` BAO method of the LeavePeriodEntitlement BAO method with an expiredOnly parameter to be added.